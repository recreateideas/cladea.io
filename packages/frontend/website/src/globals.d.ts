import { Theme } from "./App/ui-core/theme";
import { Store } from "redux";

export declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: any;
    /** enzyme global methods for testing */
    shallow: any;
    render: any;
    mount: any;
    theme: Theme;
    ResizeObserver: any;
    store: Store;
  }
}

declare module "ui-core";
