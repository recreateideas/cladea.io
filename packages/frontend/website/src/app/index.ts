export * from "./App";
export * from "./features";
