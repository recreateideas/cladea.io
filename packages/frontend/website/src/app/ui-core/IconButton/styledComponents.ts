import styled, { css } from 'styled-components';

export const Container = styled.div<{ shape?: 'square'; disabled?: boolean }>`
    ${(props) => {
        const {
            theme: {
                dsl: { palette, styles },
            },
            disabled,
            shape,
        } = props;
        const bordersRadius = shape === 'square' ? styles.borderRadius : undefined;
        return css`
            ${bordersRadius &&
            css`
                .MuiButtonBase-root {
                    border-radius: ${bordersRadius};
                }
            `}
            ${disabled &&
            css`
                &,
                * {
                    cursor: no-drop;
                    svg {
                        fill: ${palette.neutrals.inactive} !important;
                    }
                }
            `}
        `;
    }}
`;
