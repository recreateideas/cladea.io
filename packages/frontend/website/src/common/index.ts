export * from './utils';
export * from './hooks';
export * from './converters';
