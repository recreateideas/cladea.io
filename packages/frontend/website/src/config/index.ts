export const apiGatewaUrl = process.env.REACT_APP_API_GATEWAY_URL;

console.log(process.env);
