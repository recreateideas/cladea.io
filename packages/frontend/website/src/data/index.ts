import land110m from "./land-110m.json";

export { land110m };
