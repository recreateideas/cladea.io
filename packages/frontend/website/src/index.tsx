import "./app/ui-core/theme/GlobalStyle/roboto.css";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { store } from "./redux";
import { ServicedApp, ThemeProviderConnected } from "./app";

ReactDOM.render(
  <Provider store={store}>
    <ThemeProviderConnected>
      <ServicedApp />
    </ThemeProviderConnected>
  </Provider>,
  document.getElementById("root")
);
