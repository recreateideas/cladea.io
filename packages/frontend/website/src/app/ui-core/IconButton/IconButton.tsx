import { IconButton as IconButtonCore, IconButtonProps } from '@mui/material';
import { Container } from './styledComponents';
interface IProps extends IconButtonProps {
    shape?: 'square';
}

export const IconButton = (props: IProps) => {
    const { shape, ...iconButtonProps } = props;
    return (
        <Container className="icon-button" {...{ shape, disabled: iconButtonProps.disabled }}>
            <IconButtonCore {...iconButtonProps} />
        </Container>
    );
};

IconButton.displayName = 'IconButton';
