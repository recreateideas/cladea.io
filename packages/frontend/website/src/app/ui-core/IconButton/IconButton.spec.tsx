import { IconButton } from './IconButton';
import { TestProvider } from 'src/tests';

const { mountWithProviderAndTheme } = TestProvider();
describe('<IconButton/>', () => {
    it('should render', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(<IconButton>some label</IconButton>, state);
        expect(root).toMatchSnapshot();
    });
});
