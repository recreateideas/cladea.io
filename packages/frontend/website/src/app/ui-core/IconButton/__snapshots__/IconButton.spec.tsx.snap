// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton/> should render 1`] = `
Array [
  .c2 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c0::-moz-focus-inner {
  border-style: none;
}

.c0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c1 {
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0,0,0,0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c1:hover {
  background-color: rgba(0,0,0,0.04);
}

.c1.Mui-disabled {
  background-color: transparent;
  color: rgba(0,0,0,0.26);
}

@media print {
  .c0 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c1:hover {
    background-color: transparent;
  }
}

<div
    className="icon-button"
  >
    <button
      className="c0 MuiButtonBase-root c1 MuiIconButton-root MuiIconButton-sizeMedium"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      some label
      <span
        className="c2 MuiTouchRipple-root"
      />
    </button>
  </div>,
  "",
]
`;
